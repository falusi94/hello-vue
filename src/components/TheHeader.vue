<template>
  <div class="header">
    <div class="header-placeholder" />
    <div class="header-content">
      <TheHeaderButton
        title="Home"
        @click="goHome"
      />
    </div>
  </div>
</template>

<script>
import TheHeaderButton from './TheHeaderButton.vue';

export default {
  name: 'TheHeader',
  components: {
    TheHeaderButton,
  },
  methods: {
    goHome() {
      window.location = '/';
    },
  },
};
</script>

<style lang="scss" scoped>
  .header {
    height: 3.5rem;
    width: 100%;
    color: #F5F5F5;
    font-size: 1rem;
    font-weight: 800;

    &-placeholder {
      width: inherit;
      height: inherit;
    }

    &-content {
      width: inherit;
      height: inherit;
      background-color: #212121;
      position: fixed;
      top: 0;
      padding: 0 2rem;
    }
  }
</style>
