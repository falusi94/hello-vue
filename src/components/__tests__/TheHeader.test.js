import { shallowMount } from '@vue/test-utils';
import TheHeader from '../TheHeader.vue';

describe('TheHeader.vue', () => {
  it('renders properly', () => {
    const wrapper = shallowMount(TheHeader);
    expect(wrapper).toMatchSnapshot();
  });
});
