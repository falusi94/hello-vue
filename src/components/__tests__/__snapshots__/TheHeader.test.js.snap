// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader.vue renders properly 1`] = `
<div class="header">
  <div class="header-placeholder"></div>
  <div class="header-content">
    <theheaderbutton-stub title="Home"></theheaderbutton-stub>
  </div>
</div>
`;
